index.js

```diff
@@ -0,0 +1,213 @@
+const form = document.querySelector('.form');
+const textArea = document.querySelector('.search');
+const sidenavCategory = document.querySelectorAll('.category');
+const toggle = document.querySelector('#container');
+const toggleContainer = document.querySelector('#toggle-container');
+const pageTitle = document.querySelector('.page-title');
+const logo = document.querySelector('.logo');
+
+//initial API query & page render
+fetchArticles();
+
+//set 1m interval refresh
+setInterval(fetchArticles,60000);
+
+//Listener for side navigation categories
+sidenavCategory.forEach(link => {
+    link.addEventListener('click', event => {
+        modeLatestNews = true;
+        category = event.target.innerHTML;
+        APIQuery = `https://newsapi.org/v2/top-headlines?country=${country}&category=${category}&apiKey=4fede2e79736471581f9259c131f0fc2`;  
+        document.querySelector('.page-nav').setAttribute('style','display: none');
+        fetchArticles();
+        setTimeout(closeNav,500);
+        pageTitle.innerHTML = category;
+    })
+});
+
+//Listener for search query
+textArea.addEventListener('input', event => {
+    searchText = event.target.value;
+});
+
+//Listener for search submit
+form.addEventListener('submit', event => {
+    event.preventDefault();
+    modeLatestNews = false;
+    topic = searchText;
+    form.reset(); 
+    searchText = '';
+    document.querySelector(".search-box").style.display = "none";
+    APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+    document.querySelector('.page-nav').setAttribute('style','display: flex');
+    fetchArticles();
+    pageTitle.innerHTML = 'Search results for ' + topic;
+});
+
+//Listener for next page button
+pageNextButton.addEventListener('click', event => {
+    page++;
+    APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+    fetchArticles();
+});
+
+//Listener for previous page button
+pagePrevButton.addEventListener('click', event => {
+    if (page > 1) {
+        page--;
+        APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+        fetchArticles();
+    } 
+});
+
+//Listener for toggle
+toggle.addEventListener('click', function() {
+    toggleNumber = !toggleNumber;
+    if (toggleNumber) {
+        toggleContainer.style.clipPath = 'inset(0 0 0 50%)';
+        toggleContainer.style.backgroundColor = '#D74046';
+        country = 'us';
+    } else {
+        toggleContainer.style.clipPath = 'inset(0 50% 0 0)';
+        toggleContainer.style.backgroundColor = 'dodgerblue';
+        country = 'gb';
+    }
+    APIQuery = `https://newsapi.org/v2/top-headlines?country=${country}&category=${category}&apiKey=4fede2e79736471581f9259c131f0fc2`;  
```

Comment:
rather than making APIquery a global variable, I would make it local and pass it into `fetchArticles` to get data from it.

Reply:
thx, done now - although I had to remove my setInterval for 60s refresh as that relied on the global variable